src/http/web_socket/protocol.cr

```diff
@@ -233,13 +233,26 @@ class HTTP::WebSocket::Protocol
     end
   end
 
-  def close(message = nil)
+  def close(code : Int? = nil, message = nil)
     return if @io.closed?
-    if message
-      send(message.to_slice, Opcode::CLOSE)
+
+    case {code, message}
+    when {nil, nil}
+      payload = Bytes.empty
+    when {Int, nil}
+      payload = Bytes.new(2)
+      IO::ByteFormat::NetworkEndian.encode(code.to_u16, payload)
     else
-      send(Bytes.empty, Opcode::CLOSE)
+      message = message.not_nil!.to_slice
```

Comment:
There's a million ways to write this code, and all of them are a little ugly. I think I'd rather stick with this and not bikeshed.

Reply:
I agree with avoiding bikeshedding. However, when I saw that `not_nil!` I thought "Hmm... but what happens if this is actually `nil`?". Then I had to look at the code above to realize that it's probably not possible for that to not be `nil`.

If `not_nil!` was never used here this would be much easier to understand and follow. But we don't need to do it now, I can send a following PR to change it.